client/src/helpers/fhir/fhir.js

```diff
@@ -90,6 +94,37 @@ export const getObservationValue = (obs, defaultValue) => (
   get(obs, 'valueString', get(obs, 'note[0].text', defaultValue))
 )
 
+export const getNoteComment = (serviceRequest) => {
+  const value = serviceRequest?.note?.[0]?.text;
+  return value != EMPTY_NOTE_TEXT ? value : null
+}
+export const getNoteStatus = (serviceRequest) => {
+  const value = serviceRequest?.note?.[1]?.text;
+  return value != EMPTY_NOTE_TEXT ? value : null
+}
+
+export const updateNoteComment = (notes, comment) => {
```

Comment:
Beware of mutability. notes is passed by reference. Using push will mutate it. I would strongly advice using shallow copies and give back a copy.

Reply:
Agree